Geometry/HGCalCommonData/src/HGCalWaferMask.cc

```diff
@@ -1151,3 +1151,246 @@ std::vector<std::pair<double, double> > HGCalWaferMask::waferXY(
+      xy.push_back(std::make_pair((xpos + dx[np[k]]), (ypos + dy[np[k]])));
+  } else if (part == HGCalTypes::WaferLDTop) {
+    int np[12][5] = {{0, 1, 4, 5, 0},
+                     {1, 2, 5, 0, 1},
+                     {2, 3, 0, 1, 2},
+                     {3, 4, 1, 2, 3},
+                     {4, 5, 2, 3, 4},
+                     {5, 0, 3, 4, 5},
+                     {0, 1, 2, 5, 0},
+                     {5, 0, 3, 4, 5},
+                     {4, 5, 0, 3, 4},
+                     {3, 4, 5, 2, 3},
+                     {2, 3, 4, 1, 2},
+                     {1, 2, 3, 0, 1}};
+    for (int k = 0; k < 5; ++k) {
+      xy.push_back(std::make_pair((xpos + dx[np[place][k]]), (ypos + dy[np[place][k]])));
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << k << ":" << np[place][k] << ":" << dx[np[place][k]] << ":" << dy[np[place][k]];
+#endif
+    }
+  } else if (part == HGCalTypes::WaferLDBottom) {
+    int np[12][5] = {{1, 2, 3, 4, 1},
+                     {2, 3, 4, 5, 2},
+                     {3, 4, 5, 0, 3},
+                     {4, 5, 0, 1, 4},
+                     {5, 0, 1, 2, 5},
+                     {0, 1, 2, 3, 0},
+                     {5, 2, 3, 4, 5},
+                     {4, 1, 2, 3, 4},
+                     {3, 0, 1, 2, 3},
+                     {2, 5, 0, 1, 2},
+                     {1, 4, 5, 0, 1},
+                     {0, 3, 4, 5, 0}};
+    for (int k = 0; k < 5; ++k) {
+      xy.push_back(std::make_pair((xpos + dx[np[place][k]]), (ypos + dy[np[place][k]])));
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << k << ":" << np[place][k] << ":" << dx[np[place][k]] << ":" << dy[np[place][k]];
+#endif
+    }
+  } else if (part == HGCalTypes::WaferLDLeft) {
+    int np[12][6] = {{0, 1, 2, 8, 11, 0},
+                     {1, 2, 3, 9, 6, 1},
+                     {2, 3, 4, 10, 7, 2},
+                     {3, 4, 5, 11, 8, 3},
+                     {4, 5, 0, 6, 9, 4},
+                     {5, 0, 1, 7, 10, 5},
+                     {0, 6, 9, 4, 5, 0},
+                     {5, 11, 8, 3, 4, 5},
+                     {4, 10, 7, 2, 3, 4},
+                     {3, 9, 5, 1, 2, 3},
+                     {2, 8, 11, 0, 1, 2},
+                     {1, 7, 10, 5, 0, 1}};
+    for (int k = 0; k < 6; ++k) {
+      xy.push_back(std::make_pair((xpos + dx[np[place][k]]), (ypos + dy[np[place][k]])));
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << k << ":" << np[place][k] << ":" << dx[np[place][k]] << ":" << dy[np[place][k]];
+#endif
+    }
+  } else if (part == HGCalTypes::WaferLDRight) {
+    int np[12][6] = {{5, 11, 8, 3, 4, 5},
+                     {0, 6, 9, 4, 5, 0},
+                     {1, 7, 10, 5, 0, 1},
+                     {2, 8, 11, 0, 1, 2},
+                     {3, 9, 6, 1, 2, 3},
+                     {4, 10, 7, 2, 3, 4},
+                     {1, 2, 3, 9, 6, 1},
+                     {0, 1, 2, 8, 11, 0},
+                     {5, 0, 1, 7, 10, 5},
+                     {6, 5, 0, 6, 9, 4},
```

Comment:
There seems to be a genuine mistake in this definition.

Reply:
Corrected

Reply:
Done